ui/StatusQ/src/StatusQ/Core/StatusScrollView.qml

```diff
@@ -28,42 +32,51 @@ import StatusQ.Core.Utils 0.1
 
    For a list of components available see StatusQ.
 */
-Flickable {
+
+T.ScrollView {
     id: root
 
-    // NOTE: this should be replaced with margins since Qt 5.15
-    property int padding: 8
-    property int topPadding: padding
-    property int bottomPadding: padding
-    property int leftPadding: padding
-    property int rightPadding: padding
+    readonly property Flickable flickable: root.contentItem
+
+    function ensureVisible(rect) {
+        Utils.ensureVisible(flickable, rect)
+    }
+
+    function applyFlickableFix() {
+        flickable.boundsBehavior = Flickable.StopAtBounds
+        flickable.maximumFlickVelocity = 2000
+        flickable.synchronousDrag = true
+    }
 
-    readonly property int availableWidth: width - leftPadding - rightPadding
-    readonly property int availableHeight: height - topPadding - bottomPadding
+    implicitWidth: Math.max(implicitBackgroundWidth + leftInset + rightInset,
+                            contentWidth + leftPadding + rightPadding)
+    implicitHeight: Math.max(implicitBackgroundHeight + topInset + bottomInset,
+                             contentHeight + topPadding + bottomPadding)
```

Comment:
I will heat up the discussion a bit :sweat_smile: 

The main reason Qt's `Flickable`/`ScrollView` doesn't define implicit sizes is because its purpose is to provide a scrollable area, which is typically larger than its own size (as defined by width and height). Defining implicit sizes based on its content (which may be larger than the `Flickable` itself) will lead the `Flickable` to resize itself to encompass all its content, thus negating the need for scrolling. In other words, `Flickable` doesn't have natural or preferred size by definition.

On the other hand, having implicit sizes matching the content sounds very handy. Also, in most cases the size of `Flickable` can indeed be set explicitly or determined by its parent or layout, effectively fulfilling its purpose of scrolling. 

I need to think a bit further to see if it could lead to unexpected behavior.

Mentioning @micieslak, becase IIRC he has strong opinion against implicit sizes in flickable.

Reply:
You're totally right about the ‘Flickable‘. I agree on all points 🙂

But `ScrollView` does provide implicit sizes by default. The lines here that you are referring to are the default lines from any QtQuick.Controls Style.

Reply:
Any reason why not to derive directly from qq2 `ScrollView`? We would avoid assigning implicit sizes by our own.

Reply:
That's what I did at first, but for some reason Scroll Bars were duplicated.
So you would see both ours styled one and the original one at the same time 🤦

Reply:
That's super strange but I believe you. Let it be a template then :smile: